src/main/java/org/opensearch/knn/index/codec/KNN990Codec/NativeEngines990KnnVectorsWriter.java

```diff
@@ -73,6 +74,10 @@ public KnnFieldVectorsWriter<?> addField(final FieldInfo fieldInfo) throws IOExc
     @Override
     public void flush(int maxDoc, final Sorter.DocMap sortMap) throws IOException {
         flatVectorsWriter.flush(maxDoc, sortMap);
+        if (!buildVectorDatastructure) {
+            log.info("Skip building vector datastructure");
+            return;
+        }
```

Comment:
we should add a codec level tests which will ensure that if this setting is true in reality graphs are not created.

Reply:
Checking